### ISSUE TITLE
Internal Server Error: /analysis/search - Cuckoo Sandbox v2.0.1 & ElasticSearch

### ISSUE THREAD

**Issue by hakawati** [cuckoosandbox/cuckoo#1432]
Hi,
It is hard to solve the problem of linking ElasticSearch database in Cuckoo Sandbox.

- Installing JAVA
```
sudo add-apt-repository ppa:webupd8team/java
sudo apt-get update
sudo apt-get install oracle-java8-installer
sudo bash -c "echo 'JAVA_HOME=\"/usr/lib/jvm/java-8-openjdk-amd64\"' >> /etc/environment"
source /etc/environment
```

- Installing ElasticSearch
```
wget -qO - https://packages.elastic.co/GPG-KEY-elasticsearch | sudo apt-key add -
echo "deb http://packages.elastic.co/elasticsearch/2.x/debian stable main" | sudo tee -a /etc/apt/sources.list.d/elasticsearch-2.x.list
sudo apt-get update
sudo apt-get install elasticsearch
sudo systemctl daemon-reload
sudo systemctl enable elasticsearch.service
mkdir ~/ESData
sudo chown root:elasticsearch ESData
sudo chmod 777 ~/ESData
sudo usermod -a -G elasticsearch cuckoo
```
- Configuring elasticsearch.yml
```
cluster.name: es-cuckoo
node.name: es-node-n1
node.master: true
node.data: true
bootstrap.mlockall: true
path.data: /home/cuckoo/ESData
network.bind_host: 0.0.0.0
```
And then `sudo systemctl start elasticsearch`

- Configuring reporting.conf
```
[elasticsearch]
enabled = yes
hosts = 127.0.0.1:9200
timeout = 300
calls = no
index = cuckoo
index_time_pattern = yearly
cuckoo_node = es-node-n1
```
And then ~/CWD/elasticsearch/template.json into template of elasticsearch.

This is an error during Django's search at "/analysis/search/".
```
2017-04-19 22:04:36,237 [django.request] ERROR: Internal Server Error: /analysis/search/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "./analysis/views.py", line 347, in search
    "term": request.POST["search"],
  File "/usr/local/lib/python2.7/dist-packages/cuckoo/web/bin/utils.py", line 83, in render_template
    request, template_name, kwargs, status=kwargs.pop("status", 200)
  File "/usr/local/lib/python2.7/dist-packages/django/shortcuts.py", line 67, in render
    template_name, context, request=request, using=using)
  File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py", line 99, in render_to_string
    return template.render(context, request)
  File "/usr/local/lib/python2.7/dist-packages/django/template/backends/django.py", line 74, in render
    return self.template.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 209, in render
    return self._render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 201, in _render
    return self.nodelist.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 903, in render
    bit = self.render_node(node, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 917, in render_node
    return node.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/defaulttags.py", line 329, in render
    return nodelist.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 903, in render
    bit = self.render_node(node, context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/base.py", line 917, in render_node
    return node.render(context)
  File "/usr/local/lib/python2.7/dist-packages/django/template/defaulttags.py", line 224, in render
    nodelist.append(node.render(context))
  File "/usr/local/lib/python2.7/dist-packages/django/template/defaulttags.py", line 507, in render
    six.reraise(*exc_info)
  File "/usr/local/lib/python2.7/dist-packages/django/template/defaulttags.py", line 493, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 579, in reverse
    return force_text(iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs)))
  File "/usr/local/lib/python2.7/dist-packages/django/core/urlresolvers.py", line 496, in _reverse_with_prefix
    (lookup_view_s, args, kwargs, len(patterns), patterns))
NoReverseMatch: Reverse for 'analysis.views.summary' with arguments '(2,)' and keyword arguments '{}' not found. 0 pattern(s) tried: []
[19/Apr/2017 22:04:36] "POST /analysis/search/ HTTP/1.1" 500 17149
```
How do I fix it?

**Comment by jbremer** [comment]
Hi! Thanks for the bug report. We're still working on fixing up our searching capabilities (see also #1339) so for now unfortunately you won't be able to search through Cuckoo. We hope to address this somewhat soon, but it does require quite some effort from our side - so please stay tuned :-)

**Comment by jbremer** [comment]
Although we're still working on improving the search stuff, this exact issue has been resolved - see also #1408 #1410. Until we bring out the next release you'll be able to test this bug fix with version `2.0.2a2` (`pip install cuckoo==2.0.2a2`). Closing this issue as resolved. Let us know if you have any additional feedback or questions/etc.
